di.go

```diff
@@ -171,15 +171,13 @@ func RegisterBeanFactory(beanID string, beanScope Scope, beanFactory func() (int
 	initializeLock.Lock()
 	defer initializeLock.Unlock()
 	if atomic.CompareAndSwapInt32(&containerInitialized, 1, 1) {
-		return false, errors.New("container is already initialized: can't register new bean")
+		return false, errors.New("container is already initialized: can't register new bean factory")
 	}
-	var existingBeanType reflect.Type
 	var ok bool
-	if existingBeanType, ok = beans[beanID]; ok {
+	if _, ok = beanFactories[beanID]; ok {
```

Comment:
This is not what I wanted to do here :) I didn't want to check if there's already such a bean factory. I wanted to check if there's already such a bean. Check the comment:

```
// RegisterBeanFactory function registers bean, provided the bean factory that will be used by the container in order to
// create an instance of this bean. `beanScope` can be any scope of the supported ones. `beanFactory` can only produce a
// reference or an interface. Return value of `overwritten` is set to `true` if the bean with the same `beanID` has been
// registered already.
```

Reply:
`RegisterBeanFactory function registers bean`

I read this as 'bean factor registered as bean itself', that is why all next rows for me addressed to current factory as bean and i fixed it in that way.

I reverted my changes and changed tests accordingly. 
However there is something odd in this logic.
1. Beans map is not cleared on override, do we really need to store it? How to understand what map is the primary source of the data?
1. Can we register bean factory first and then try to registerBean, what is gonna happen after? (Both maps beans and beanFactories will contain information about  bean with the same id.
 1. Plus message could  be improved  with direct information that happened overwrite from Bean to BeanFactory.